File: exercise/java/day11/src/main/java/christmas/Preparation.java
package christmas;

public class Preparation {
    private Preparation() {
    }

    public static String prepareGifts(int numberOfGifts) {
        if (numberOfGifts <= 0) {
            return "No gifts to prepare.";
        } else if (numberOfGifts < 50) {
            return "Elves will prepare the gifts.";
        }
        return "Santa will prepare the gifts.";
    }

    public static String categorizeGift(int age) {
        if (age <= 2) {
            return "Baby";
        } else if (age <= 5) {
            return "Toddler";
        } else if (age <= 12) {
            return "Child";
        }
        return "Teen";
    }

    public static boolean ensureToyBalance(ToyType toyType, int toysCount, int totalToys) {
        double typePercentage = (double) toysCount / totalToys;

        return switch (toyType) {
            case EDUCATIONAL -> typePercentage >= 0.25;
            case FUN -> typePercentage >= 0.30;
            case CREATIVE -> typePercentage >= 0.20;
        };
    }
}

File: exercise/kotlin/day11/src/main/kotlin/christmas/Preparation.kt
package christmas

import christmas.ToyType.*

enum class ToyType {
    EDUCATIONAL, FUN, CREATIVE
}

object Preparation {
    fun prepareGifts(numberOfGifts: Int): String = when {
        numberOfGifts <= 0 -> "No gifts to prepare."
        numberOfGifts < 50 -> "Elves will prepare the gifts."
        else -> "Santa will prepare the gifts."
    }

    fun categorizeGift(age: Int): String = when {
        age <= 2 -> "Baby"
        age <= 5 -> "Toddler"
        age <= 12 -> "Child"
        else -> "Teen"
    }

    fun ensureToyBalance(toyType: ToyType, toysCount: Int, totalToys: Int): Boolean =
        (toysCount.toDouble() / totalToys)
            .let { typePercentage ->
                return when (toyType) {
                    EDUCATIONAL -> typePercentage >= 0.25
                    FUN -> typePercentage >= 0.30
                    CREATIVE -> typePercentage >= 0.20
                }
            }
}